### ISSUE TITLE
A question about the way of how to split data

### ISSUE THREAD

**Issue by ConanCui** [dawenl/vae_cf#2]
Hi, I read your paper and run your experiment code in detail, it is a really nice work. Now I am doing some experiment and I have some question about how to split the data. In your paper and code,  the user is split into training/validation/test sets, and train the model using the entire history of training users, and when evaluate, the model take part of the history from held-out users.

But Neural collaborative filtering(NCF) split the data in (user,item) level, I have question about how you split the data for NCF in user level? Cause NCF learn embedding for different users from the data. I guess that when training maybe we feed the entire history of training user to train the model and get the embedding of users. When testing, we use part of the history of test user to learn the embedding of the test user, and use the embedding to predict the rest of the history of test user. But the embedding of training user can't have any influence on test users, so  is the training user set  pointless? 

The data split method in your paper is different from the  recommendation papers I have read, mostly the split is do like that, we use part of history of all the users to learn the embedding of all users, and evaluate it on the rest history of all users. So I have some question here.

And I am looking forward your reply sincerely.

**Comment by dawenl** [comment]
As you can see in the paper, for NCF we didn't use the same evaluation protocol, instead we used the data the original authors provided, which splits the data in (user, item) level. 

It is possible to do it on the user-level just as you said -- the user embeddings from the training users is not technically useless, since they are necessary for you to learn the item embeddings, which will be shared for both training and evaluation.

**Comment by ConanCui** [comment]
Thanks for your reply. The second protocol you said I can understand.

Do you use the first protocol you said for NCF to train your model, I mean when you compare your model with NCF, dose your model VAE-CF also use the protocol that split the data in (use, item) level? 

If we do the split like this, some history of all the user is used to train, and rest history of all the user is used to test. But I think this training protocol will harm the performance of your model. Cause we only provide the VAE-CF model part history, how can we expect it will reconstruct the entire history.  And this is different from that we use drop out in input layer, although we use drop out in input layer, we still compute the reconstruction loss between prediction and entire history,  but if we use the protocol(user-item) level to train the VAE-CF model, that means the model can only compute the loss between prediction and part history.

Is there anything I misunderstand, I hope our discussion will also help other people, and this will not cost you too much time. And I am still looking forward your reply sincerely.

**Comment by dawenl** [comment]
Yes, on NCF I always use (user, item) split -- otherwise, how are different numbers comparable? 

I don't see why this could be a disadvantage to VAE-CF. The whole point of applying dropout is to hope the model able to reconstruct the full history (in the training set) from only part of the history. And by doing that, you would hope the model can also generalize to the test set, assuming both dropout and test set selection are performed randomly.

**Comment by ConanCui** [comment]
I swear this is the last question.:joy:

In short, when compare VAE-CF with NCF, the training data we feed to VAE-CF is sampled in (user,item) level and test VAE-CF with the test data sampled in (user,item) as well. Is this right？

**Comment by dawenl** [comment]
In case there is any confusion, [this](https://github.com/hexiangnan/neural_collaborative_filtering/tree/master/Data) is the data i used (both train and test) for VAE-CF -- part of the train is excluded as a validation set.
